[BUG] Incorrect export of types and interfaces

# Problem
At the moment we're having reusable SFCs with `<script setup lang="ts">` with wrongly exported types _and/or_ interfaces. Interfaces _and/or_ types exported from `.vue` and imported to `.ts` files lead to errors while running `check:types` script - `tsc` and `vue-tsc`.

Here's an example:
```bash
> @infermedica/component-library@0.5.2 check:types /component-library
> tsc --noEmit & vue-tsc --noEmit

src/components/organisms/UiPhoneNumber/UiPhoneNumber.stories.ts:2:15 - error TS2614: Module '"*.vue"' has no exported member 'UiPhoneNumberProps'. Did you mean to use 'import UiPhoneNumberProps from "*.vue"' instead?

2 import type { UiPhoneNumberProps } from '@/components/organisms/UiPhoneNumber/UiPhoneNumber.vue';
                ~~~~~~~~~~~~~~~~~~


Found 1 error in src/components/organisms/UiPhoneNumber/UiPhoneNumber.stories.ts:2
```

## Solution
We have several ways to fix this. Regarding many sources we could either:
- create another file in each of the component's directories and export component-related types from those files; these files could be named like:
  - `types.ts` **_[preferred]_**
  - or `ComponentNameTypes.ts`
  - or `componentNameTypes.ts`
- add another, parallel `script` tag aside current one, but without `setup` and only export from within those tags

--- a/package.json
+++ b/package.json
@@ -12,7 +12,7 @@
     "test:unit": "vitest --run",
     "test:watch": "vitest -w --changed",
     "lint": "eslint --fix --ext .js,.vue,.ts src",
-    "check:types": "tsc --noEmit & vue-tsc --noEmit",
+    "check:types": "vue-tsc --noEmit",
     "stylelint": "stylelint src/**/*.{vue,scss,css} --fix",
     "storybook": "start-storybook -p 6006",
     "build-storybook": "build-storybook",

